components/brave_rewards/browser/rewards_service_impl_unittest.cc

```diff
@@ -78,8 +79,8 @@ class RewardsServiceTest : public testing::Test {
     auto* rewards_ = new RewardsServiceImpl(profile());
 #endif
     RewardsServiceFactory::SetServiceForTesting(std::move(rewards_));
-    rewards_service_ = static_cast<RewardsServiceImpl*>(
-        RewardsServiceFactory::GetForProfile(profile()));
+    rewards_service_ = base::WrapUnique(static_cast<RewardsServiceImpl*>(
```

Comment:
This whole thing seems strange. What's the point of getting `rewards_service_` from the factory when we set `rewards_` as the service to be returned by the factory? It seems like `SetServiceForTesting` should just go away entirely and either use the directly created `rewards_service_` or use the one returned by `RewardsServiceFactory::GetForProfile` and not do https://github.com/brave/brave-core/pull/20306/files#diff-79a8ff13d4400b7fb9810d76aa14728acb0aded37d2dc1c414a8b00396489642R77 at all

Reply:
Fixed a21f6b7c5d5353cc933ca9ecf3eac77e26d727ab.